.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -0,0 +1,151 @@
+name: Deploy Sphinx documentation
+
+on:
+
+  workflow_call:
+    inputs:
+        python-version:
+          required: true
+          type: string
+        package-manager:
+          required: true 
+          type: string
+        dependency-path:
+          required: true 
+          type: string
+        trigger-label:
+          required: false
+          type: string
+          default: none
+        path-doc:
+          required: false
+          type: string 
+          default: ./doc
+
+        
+          
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  check-label:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.valid_label.outputs.value}}
+    steps:
+      - name: checks for the label 
+        id: valid_label
+        run: |
+          if [[ "${{ inputs.trigger-label }}" == "latest" ]] || [[ "${{ inputs.trigger-label  }}" == "stable" ]];
+            then
+                echo "value=true" >> $GITHUB_OUTPUT
+            else
+                echo "value=false" >> $GITHUB_OUTPUT 
+          fi
+            
+  
+  build:
+    
+    runs-on: ubuntu-latest
+    needs: check-label
+    if: needs.check-label.outputs.label == 'true'
+    steps:
+      
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+        if: ${{ inputs.package-manager == 'poetry' }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+          cache: ${{ inputs.package-manager }}
+          cache-dependency-path: ${{ inputs.dependency-path }} 
+      - name: Install dependencies with pip
+        run: |
+          ${{ inputs.package-manager }} install .[docs]
```

Comment:
Since you are already checking if it's `pip`, then you do not need the variable here.

In the end, it is more readable `pip install .[docs]`, and more explicit (since they are equivalent).